From ab8980973866a096cf15b91ebd21053e473141ca Mon Sep 17 00:00:00 2001
From: Jostar Yang <jostar_yang@accton.com.tw>
Date: Fri, 5 Feb 2021 15:08:48 +0800
Subject: [PATCH] [as4630-54pe] Fix fan LED always showing fan is turned off

1. The board FAN/PSU led is controlled by CPLD
2. Modify code to let FAN/PSU LED show mode=auto

Signed-off-by: Jostar Yang <jostar_yang@accton.com.tw>
---
 .../builds/x86-64-accton-as4630-54pe-leds.c   | 46 ++++---------------
 .../module/src/ledi.c                         | 19 ++------
 2 files changed, 13 insertions(+), 52 deletions(-)

diff --git a/packages/platforms/accton/x86-64/as4630-54pe/modules/builds/x86-64-accton-as4630-54pe-leds.c b/packages/platforms/accton/x86-64/as4630-54pe/modules/builds/x86-64-accton-as4630-54pe-leds.c
index 04caf13e..ff5d01c5 100755
--- a/packages/platforms/accton/x86-64/as4630-54pe/modules/builds/x86-64-accton-as4630-54pe-leds.c
+++ b/packages/platforms/accton/x86-64/as4630-54pe/modules/builds/x86-64-accton-as4630-54pe-leds.c
@@ -362,40 +362,14 @@ static enum led_brightness accton_as4630_54pe_led_stk2_get(struct led_classdev *
     return led_reg_val_to_light_mode(LED_TYPE_STK2, ledctl->reg_val[1]);
 }
 
-static void accton_as4630_54pe_led_fan_set(struct led_classdev *led_cdev,
-        enum led_brightness led_light_mode)
-{
-    accton_as4630_54pe_led_set(led_cdev, led_light_mode, LED_TYPE_FAN);
-}
-
-static enum led_brightness accton_as4630_54pe_led_fan_get(struct led_classdev *cdev)
-{
-    accton_as4630_54pe_led_update();
-    return led_reg_val_to_light_mode(LED_TYPE_FAN, ledctl->reg_val[0]);
-}
-
-static void accton_as4630_54pe_led_psu1_set(struct led_classdev *led_cdev,
-        enum led_brightness led_light_mode)
+static void as4630_54pe_led_auto_set(struct led_classdev *led_cdev,
+                                           enum led_brightness led_light_mode)
 {
-    accton_as4630_54pe_led_set(led_cdev, led_light_mode, LED_TYPE_PSU1);
 }
 
-static enum led_brightness accton_as4630_54pe_led_psu1_get(struct led_classdev *cdev)
+static enum led_brightness as4630_54pe_led_auto_get(struct led_classdev *cdev)
 {
-    accton_as4630_54pe_led_update();
-    return led_reg_val_to_light_mode(LED_TYPE_PSU1, ledctl->reg_val[0]);
-}
-
-static void accton_as4630_54pe_led_psu2_set(struct led_classdev *led_cdev,
-        enum led_brightness led_light_mode)
-{
-    accton_as4630_54pe_led_set(led_cdev, led_light_mode, LED_TYPE_PSU2);
-}
-
-static enum led_brightness accton_as4630_54pe_led_psu2_get(struct led_classdev *cdev)
-{
-    accton_as4630_54pe_led_update();
-    return led_reg_val_to_light_mode(LED_TYPE_PSU2, ledctl->reg_val[0]);
+    return LED_MODE_AUTO;
 }
 
 static struct led_classdev accton_as4630_54pe_leds[] = {
@@ -442,24 +416,24 @@ static struct led_classdev accton_as4630_54pe_leds[] = {
     [LED_TYPE_FAN] = {
         .name			 = "fan",
         .default_trigger = "unused",
-        .brightness_set	 = accton_as4630_54pe_led_fan_set,
-        .brightness_get  = accton_as4630_54pe_led_fan_get,
+        .brightness_set	 = as4630_54pe_led_auto_set,
+        .brightness_get  = as4630_54pe_led_auto_get,
         .flags			 = LED_CORE_SUSPENDRESUME,
         .max_brightness  = LED_MODE_AUTO,
     },
     [LED_TYPE_PSU1] = {
         .name			 = "psu1",
         .default_trigger = "unused",
-        .brightness_set	 = accton_as4630_54pe_led_psu1_set,
-        .brightness_get  = accton_as4630_54pe_led_psu1_get,
+        .brightness_set	 = as4630_54pe_led_auto_set,
+        .brightness_get  = as4630_54pe_led_auto_get,
         .flags			 = LED_CORE_SUSPENDRESUME,
         .max_brightness  = LED_MODE_AUTO,
     },
     [LED_TYPE_PSU2] = {
         .name			 = "psu2",
         .default_trigger = "unused",
-        .brightness_set	 = accton_as4630_54pe_led_psu2_set,
-        .brightness_get  = accton_as4630_54pe_led_psu2_get,
+        .brightness_set	 = as4630_54pe_led_auto_set,
+        .brightness_get  = as4630_54pe_led_auto_get,
         .flags			 = LED_CORE_SUSPENDRESUME,
         .max_brightness  = LED_MODE_AUTO,
     },
diff --git a/packages/platforms/accton/x86-64/as4630-54pe/onlp/builds/x86_64_accton_as4630_54pe/module/src/ledi.c b/packages/platforms/accton/x86-64/as4630-54pe/onlp/builds/x86_64_accton_as4630_54pe/module/src/ledi.c
index 6cd09d87..aa115a74 100755
--- a/packages/platforms/accton/x86-64/as4630-54pe/onlp/builds/x86_64_accton_as4630_54pe/module/src/ledi.c
+++ b/packages/platforms/accton/x86-64/as4630-54pe/onlp/builds/x86_64_accton_as4630_54pe/module/src/ledi.c
@@ -82,32 +82,19 @@ led_light_mode_map_t led_map[] = {
 {LED_DIAG, LED_MODE_GREEN,       ONLP_LED_MODE_GREEN},
 {LED_DIAG, LED_MODE_AMBER,       ONLP_LED_MODE_ORANGE},
 {LED_DIAG, LED_MODE_GREEN_BLINK, ONLP_LED_MODE_GREEN_BLINKING},
-
 {LED_PRI, LED_MODE_OFF,     ONLP_LED_MODE_OFF},
 {LED_PRI, LED_MODE_AMBER,   ONLP_LED_MODE_ORANGE},
 {LED_PRI, LED_MODE_GREEN,   ONLP_LED_MODE_GREEN},
-
 {LED_POE, LED_MODE_OFF,     ONLP_LED_MODE_OFF},
 {LED_POE, LED_MODE_AMBER,   ONLP_LED_MODE_ORANGE},
 {LED_POE, LED_MODE_GREEN,   ONLP_LED_MODE_GREEN},
-
 {LED_STK1, LED_MODE_OFF,    ONLP_LED_MODE_OFF},
 {LED_STK1, LED_MODE_GREEN,  ONLP_LED_MODE_GREEN},
-
 {LED_STK2, LED_MODE_OFF,    ONLP_LED_MODE_OFF},
 {LED_STK2, LED_MODE_GREEN,  ONLP_LED_MODE_GREEN},
-
-{LED_FAN, LED_MODE_OFF,    ONLP_LED_MODE_OFF},
-{LED_FAN, LED_MODE_AMBER,  ONLP_LED_MODE_GREEN},
-{LED_FAN, LED_MODE_GREEN,  ONLP_LED_MODE_ORANGE},
-
-{LED_PSU1, LED_MODE_OFF,   ONLP_LED_MODE_OFF},
-{LED_PSU1, LED_MODE_AMBER, ONLP_LED_MODE_GREEN},
-{LED_PSU1, LED_MODE_GREEN, ONLP_LED_MODE_ORANGE},
-
-{LED_PSU2, LED_MODE_OFF,   ONLP_LED_MODE_OFF},
-{LED_PSU2, LED_MODE_AMBER, ONLP_LED_MODE_GREEN},
-{LED_PSU2, LED_MODE_GREEN, ONLP_LED_MODE_ORANGE}
+{LED_FAN, LED_MODE_AUTO,    ONLP_LED_MODE_AUTO},
+{LED_PSU1, LED_MODE_AUTO,   ONLP_LED_MODE_AUTO},
+{LED_PSU2, LED_MODE_AUTO,   ONLP_LED_MODE_AUTO}
 };
 
 static char last_path[][10] =  /* must map with onlp_led_id */
-- 
2.31.1

